src/librustc_mir/borrow_check/mutability_errors.rs

```diff
@@ -41,14 +41,19 @@ impl<'a, 'gcx, 'tcx> MirBorrowckCtxt<'a, 'gcx, 'tcx> {
         );
 
         let mut err;
-        let item_msg;
         let reason;
         let access_place_desc = self.describe_place(access_place);
         debug!("report_mutability_error: access_place_desc={:?}", access_place_desc);
 
+        let mut item_msg = match (&access_place_desc, &the_place_err) {
+            (Some(desc), _) => format!("`{}`", desc),
+            (None, Place::Base(PlaceBase::Local(local))) if self.mir.local_decls[*local]
+                .source_info.span.is_compiler_desugaring(CompilerDesugaringKind::Async)
+            => "async `fn` parameter".to_string(),
```

Comment:
```suggestion
            => "`async fn` parameter".to_string(),
```

I can’t remember if there’s precedent for how we quote `async fn`.
(Somehow managed to accidentally comment this in the commit first, oops)

Reply:
Done